### PR TITLE
Constrain service attach point

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
Constraint annotation attach-point `service` only for `service-declarations` and `service constructor expression`s.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29073

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangClassDefinition.java [hunk 1]
```diff
@@ -55,6 +55,7 @@
     public List<BLangSimpleVariable> referencedFields;
     public int precedence;
     public boolean isServiceDecl;
+    public boolean isObjectConstructor;
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangClassDefinition.java:58]: We already have an `OBJECT_CTOR` flag, right? Can't we reuse the same without adding another field?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentNegativeTest.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
     @BeforeClass
     public void setup() {
         compileResult = BCompileUtil.compile("test-src/annotations/annot_attachments_negative.bal");
-        Assert.assertEquals(compileResult.getErrorCount(), 253);
+        Assert.assertEquals(compileResult.getErrorCount(), 266);
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentNegativeTest.java:40]: Do we already have positive tests for this?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangClassDefinition.java (hunk 1 above)
Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangClassDefinition.java:58]: Oh totally missed that, will use it

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentNegativeTest.java (hunk 2 above)
Comment (rdhananjaya) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentNegativeTest.java:40]: Added

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**